### PR TITLE
Block Edit: use hooks

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Simplifies the code. There's no noticeable change in performance. Measured a few times and typing time seems slightly lower.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-edit/index.js [hunk 1]
```diff
@@ -1,69 +1,39 @@
...
-		// It is important to return the same object if props haven't changed
-		// to avoid  unnecessary rerenders.
-		// See https://reactjs.org/docs/context.html#caveats.
-		this.propsToContext = memize( this.propsToContext.bind( this ), {
-			maxSize: 1,
-		} );
-	}
-
-	propsToContext(
+function BlockEdit( props ) {
```
Comment (ZebulanStanphill) [inline packages/block-editor/src/components/block-edit/index.js:12]: We could use destructuring here in the function parameters, as we do for other React components.

Also, we can do `export default function ...` here.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: packages/block-editor/src/components/block-edit/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/components/block-edit/index.js:12]: We need `props`.

Comment (ZebulanStanphill) [reply packages/block-editor/src/components/block-edit/index.js]: Ah, I see; I missed that.

**Pull request merged**